maltoolbox/model.py

```diff
@@ -864,4 +722,186 @@ def load_from_file(
             serialized_model = load_dict_from_json_file(filename)
         else:
             raise ValueError('Unknown file extension, expected json/yml/yaml')
-        return cls._from_dict(serialized_model, lang_classes_factory)
+        return cls._from_dict(serialized_model, lang_graph)
+
+
+class ModelAsset:
+    def __init__(self, name:str, lg_asset: LanguageGraphAsset):
+        self.name:str = name
+        self.id:Optional[int] = None
+        self.lg_asset:LanguageGraphAsset = lg_asset
+        self.type = self.lg_asset.name
+        self.defenses:dict = {}
+        self.extras:dict = {}
+        self.associations:list = []
+        self.attack_step_nodes:list = []
+        for step in self.lg_asset.attack_steps.values():
+            if step.type == 'defense':
+                self.defenses[step.name] = 1.0 if step.ttc and \
+                    step.ttc['name'] == 'Enabled' else 0.0
+
+
```

Comment:
Making this a dataclass will simplify the code and give access to ready implementations of `__repr__` and `__str__`, and also support `asdict` that could simplify `_to_dict` in the future.

Reply:
Not happening, I've had my full of dataclasses. :)

Reply:
@mrkickling and I decided to not have it be a dataclass just yet because of the default defense values handling. We might change it to a dataclass down the line if we feel it's a particularly good fit.